src/upload/UploadPanel.stories.tsx

```diff
@@ -119,9 +119,11 @@ withItems.story = {
             await editActionPromise;
 
             const doneActionPromise = interceptAction(page, "uploadDone");
+            const editTimeoutPromise = interceptActionWithTimeout(page, "uploadEditMetadata", 500);
```

Comment:
Hmm, this structure gives me an interesting idea that I think is better than a timeout.
We can race the `interceptAction` promise with a promise that we resolve immediately after `await doneButton.click()`.
A promise is never resolved synchronously, so it will always happen after we had time to receive the triggered action.

If the action was triggered, that promise will win the race. Else, the promise we created after the click will win the race.

That way, we can get rid of timeouts. The problem with timeouts is that they delay the tests when they are too long, and that they result in broken tests when they are too short. You now guessed 500ms to be a good time, but you don't know how long your browser will actually take to perform a click.

Reply:
That does sound much better! I felt using timeouts was a bit iffy, but couldn't think of a better way.